packages/x-data-grid/src/hooks/features/preferencesPanel/gridPreferencePanelSelector.ts

```diff
@@ -7,7 +7,7 @@ export const gridPreferencePanelStateSelector = createRootSelector(
 
 export const gridPreferencePanelSelectorWithLabel = createSelector(
   gridPreferencePanelStateSelector,
-  (panel, labelId: string) => {
+  (panel, labelId: string | undefined) => {
```

Comment:
The typings from the base-ui selector functions are more precise, they can catch more issues.